Update trim-self-contained.md (#20278)
dotnet/docs: docs/core/deploying/trim-self-contained.md

However, there is a risk that the build time analysis of the application can cause failures at runtime, due to not being able to reliably analyze various problematic code patterns (largely centered on reflection use). Because reliability can't be guaranteed, this deployment model is offered as a preview feature. 

The build time analysis engine provides warnings to the developer of code patterns that are problemmatic to detect which other code is required. Code can be annotated with attributes to tell the trimmer what else to include. Many reflection patterns can be replaced with build-time code generation using [Source Generators](https://github.com/dotnet/roslyn/blob/master/docs/features/source-generators.md).

> The trim mode for the applications can be configured via the TrimMode and will default (`copyused`) to bundle assemblies that are used in the application. Blazor WebAssembly applications will use a more aggressive mode (`trim`) that will trim unused code within assemblies. Trim analysis warnings give information on code patterns where a full dependency analysis was not possible. These warnings are suppressed by default and can be turned on by setting the flag, `SuppressTrimAnalysisWarnings`, to false. More information on the trim options available can be found at the [ILLinker page](https://github.com/mono/linker/blob/master/docs/illink-options.md).

Reviewer: Minor wording adjustments. We generally don't put a value in ``(`value`)`` syntax and instead call it out directly.

```suggestion
The trim mode for the applications is configured with the `TrimMode` setting. The default value is `copyused` and bundles referenced assemblies with the application. The `trim` value is used with Blazor WebAssembly applications and trims unused code within assemblies. Trim analysis warnings give information on code patterns where a full dependency analysis was not possible. These warnings are suppressed by default and can be turned on by setting the flag `SuppressTrimAnalysisWarnings` to `false`. For more information about the trim options available, see the [ILLinker page](https://github.com/mono/linker/blob/master/docs/illink-options.md).
```
